interfaces.go

```diff
@@ -48,7 +48,9 @@ type DirectedGraph[NodeType any] interface {
 	// resolved or unresolvable, and clears the list.
 	// A node becomes ready to process when all of its AND dependencies and at least one of
 	// its OR dependencies are resolved.
-	PopReadyNodes() []*node[NodeType]
+	PopReadyNodes() map[string]ResolutionStatus
+	// HasReadyNodes Checks to see if there are any ready nodes without clearing them.
+	HasReadyNodes() bool
```

Comment:
In the other header comments here, the second word  (here, "Checks") is not capitalized.

Reply:
Fixed.